File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    arr = []
    for i in range(len(ingredient)):
        if len(arr) == 0:
            arr.append(ingredient[i])
        else:
            if arr[len(arr)-1] == 1 and ingredient[i] == 2:
                arr.pop()
                arr.append(12)
            elif arr[len(arr)-1] == 12 and ingredient[i] == 3:
                arr.pop()
                arr.append(123)
            elif arr[len(arr)-1] == 123 and ingredient[i] == 1:
                arr.pop()
                answer += 1
            else:
                arr.append(ingredient[i])
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        int count = 0;
        for(int i=0; i<ingredient.length; i++){
            if(stack.isEmpty()){
                stack.push(ingredient[i]);
            } else {
                if(stack.peek() == 1 && ingredient[i] == 2){
                    stack.pop();
                    stack.push(12);
                } else if(stack.peek() == 12 && ingredient[i] == 3){
                    stack.pop();
                    stack.push(123);
                } else if(stack.peek() == 123 && ingredient[i] == 1){
                    stack.pop();
                    answer++;
                } else {
                    stack.push(ingredient[i]);
                }
            }
        }
        return answer;
    }
}